File: Print All Leaf Nodes of a Binary Tree from Right to Left/Tree.cpp
//
// Created by MRIGANKA on 3/20/2022.
//

#include "Tree.h"

void Tree::print_leaf_nodes(Node *root) {
    if (!root) {
        return;
    }

    if (!root->left && !root->right) {
        std::cout << root->data << " ";
    }

    print_leaf_nodes(root->right);
    print_leaf_nodes(root->left);
}

File: Left View of Binary Tree/src/com/Releationship/Tree.java
package com.Releationship;

public class Tree {
    public Node root;
    public int max_level = 0;

    public void inOrder() {
        inOrder(root);
    }

    public void inOrder(Node root) {
        if (root == null) {
            return;
        }
        inOrder(root.left);
        System.out.println(root.data);
        inOrder(root.right);
    }

    public void leftView() {
        int level = 1;
        leftView(root, level);
    }

    public void leftView(Node root, int level) {
        if (root == null) {
            return;
        }
        if (max_level < level) {
            System.out.println(root.data);
            max_level = level;
        }
        leftView(root.left, level + 1);
        leftView(root.right, level + 1);
    }
}
